src/AzureIoTHub.Portal.Server/Startup.cs

```diff
@@ -325,6 +306,41 @@ public void ConfigureServices(IServiceCollection services)
             _ = services.AddQuartzHostedService(q => q.WaitForJobsToComplete = true);
         }
 
+        private static void ConfigureServicesAzure(IServiceCollection services)
+        {
+            _ = services.AddTransient<IExportManager, ExportManager>();
+            _ = services.AddTransient<IExternalDeviceService, ExternalDeviceService>();
+            _ = services.AddTransient<IConfigService, ConfigService>();
+            _ = services.AddTransient<IDeviceTagService, DeviceTagService>();
+            _ = services.AddTransient<ILoRaWANCommandService, LoRaWANCommandService>();
+            _ = services.AddTransient<IEdgeModelService, EdgeModelService>();
+            _ = services.AddTransient<ILoRaWANConcentratorService, LoRaWANConcentratorService>();
+            _ = services.AddTransient<IEdgeDevicesService, EdgeDevicesService>();
+            _ = services.AddTransient<IDevicePropertyService, DevicePropertyService>();
+            _ = services.AddTransient<IDeviceConfigurationsService, DeviceConfigurationsService>();
+            _ = services.AddTransient<IDeviceModelPropertiesService, DeviceModelPropertiesService>();
+            _ = services.AddTransient(typeof(IDeviceModelService<,>), typeof(DeviceModelService<,>));
+            _ = services.AddTransient<IDeviceService<DeviceDetails>, DeviceService>();
+            _ = services.AddTransient<IDeviceService<LoRaDeviceDetails>, LoRaWanDeviceService>();
+        }
+
+        private static void ConfigureServicesAWS(IServiceCollection services, ConfigHandler configuration)
+        {
+            _ = services.AddSingleton(() => new AmazonIoTClient(configuration.AWSAccess, configuration.AWSAccessSecret, RegionEndpoint.GetBySystemName(configuration.AWSRegion)));
+            _ = services.AddSingleton(async sp =>
+            {
+                var endpoint = await sp.GetService<AmazonIoTClient>().DescribeEndpointAsync(new Amazon.IoT.Model.DescribeEndpointRequest
+                {
+                    EndpointType = "iot:Data-ATS"
+                });
+
+                return new AmazonIotDataClient(configuration.AWSAccess, configuration.AWSAccessSecret, new AmazonIotDataConfig
+                {
+                    ServiceURL = $"https://{endpoint.EndpointAddress}"
+                });
+            });
```

Comment:
To my mind, these should be moved in the ``AWSServiceCollectionExtension`` because they are part of the infrastructure layer.
@hocinehacherouf ?

Reply:
Yes absolutely, I saw with Hocine and I made the change.